impl/src/main/java/com/okta/sdk/impl/util/OAuth2Utils.java

```diff
@@ -0,0 +1,91 @@
+import java.io.FileReader;
+import java.io.IOException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.UUID;
+
+public class OAuth2Utils {
+    private static final Logger log = LoggerFactory.getLogger(OAuth2Utils.class);
+
+    /**
+     * Create signed JWT string with the supplied {@link ClientConfiguration} details.
+     *
+     * @param {@link ClientConfiguration} object
+     * @return signed JWT string
+     * @throws {@link InvalidKeySpecException}
+     * @throws {@link NoSuchAlgorithmException}
+     * @throws {@link IOException}
+     */
+    public static String createSignedJWT(ClientConfiguration clientConfiguration)
+        throws InvalidKeySpecException, NoSuchAlgorithmException, IOException {
+        String clientId = clientConfiguration.getClientId();
+        SignatureAlgorithm signatureAlgorithm = SignatureAlgorithm.forName(clientConfiguration.getAlgorithm());
+        PrivateKey privateKey = createPrivateKeyFromPemFilePath(clientConfiguration.getKeyFilePath());
+        Instant now = Instant.now();
+
+        String jwt = Jwts.builder()
+            .setAudience(clientConfiguration.getBaseUrl() + "/oauth2/v1/token")
+            .setIssuedAt(Date.from(now))
+            .setExpiration(Date.from(now.plus(1L, ChronoUnit.HOURS)))
+            .setIssuer(clientId)
+            .setSubject(clientId)
+            .claim("jti", UUID.randomUUID().toString())
+            .signWith(signatureAlgorithm, privateKey)
+            .compact();
+
+        return jwt;
+    }
+
+    /**
+     * Create {@link PrivateKey} instance from a given pem key file path.
+     *
+     * @param pemFilePath string
+     * @return {@link PrivateKey} instance
+     * @throws {@link IOException}
+     * @throws {@link InvalidKeySpecException} if private key is not RSA or EC based.
+     * @throws {@link NoSuchAlgorithmException} if an unsupported algorithm is detected.
+     */
+    static PrivateKey createPrivateKeyFromPemFilePath(final String pemFilePath)
+        throws IOException, InvalidKeySpecException, NoSuchAlgorithmException {
+        log.debug("Creating private key from pem file {}", pemFilePath);
+        File privateKeyFile = new File(pemFilePath);
+        if (!privateKeyFile.exists()) {
+            throw new FileNotFoundException(pemFilePath);
+        }
+
+        PemReader pemReader = new PemReader(new FileReader(pemFilePath));
+        PemObject pemObject = pemReader.readPemObject();
+        byte[] pemContent = pemObject.getContent();
+        pemReader.close();
+        PKCS8EncodedKeySpec encodedKeySpec = new PKCS8EncodedKeySpec(pemContent);
+
+        try {
+            return KeyFactory.getInstance("RSA").generatePrivate(encodedKeySpec);
```

Comment:
I linked to a StackOverflow comment that shows a better way to do this.  You shouldn't need to call `generatePrivate` at all. Ping me if you cannot find it

Reply:
https://stackoverflow.com/questions/11787571/how-to-read-pem-file-to-get-private-and-public-key/60109600#60109600

Reply:
Noted. This would require pulling in bcpix library. If that is not a concern here, I'd be ok to follow that.